src/main/java/com/gitblit/auth/LdapAuthProvider.java

```diff
@@ -527,12 +527,35 @@ private TeamModel createTeamFromLdap(SearchResultEntry teamEntry) {
 		return answer;
 	}
 
-	private SearchResult doSearch(LDAPConnection ldapConnection, String base, String filter) {
+	private List<SearchResultEntry> doSearch(LDAPConnection ldapConnection, String base, String filter) {
 		try {
-			return ldapConnection.search(base, SearchScope.SUB, filter);
-		} catch (LDAPSearchException e) {
-			logger.error("Problem Searching LDAP", e);
+			// Perform a search to retrieve all users in the server, but only retrieving
+			// ten at a time.
+			List<SearchResultEntry> foundResults = new LinkedList<SearchResultEntry>();
+			SearchRequest searchRequest = new SearchRequest(base, SearchScope.SUB, filter);
+			ASN1OctetString resumeCookie = null;
+
+			while (true) {
+				searchRequest.setControls(new SimplePagedResultsControl(100, resumeCookie));
```

Comment:
Best would be to either:
* Make `100` a constant so that it can be easily read, understood, and changed later on
* Or (even better): make this configurable in `gitblit.properties`